hi-ml-histopathology/testhisto/testhisto/models/test_deepmil.py

```diff
@@ -208,41 +221,12 @@ def test_metrics() -> None:
 
     for key, metric_obj in module_metrics_dict.items():
         value = metric_obj.compute()
-        expected_value = independent_metrics_dict[key](predicted_probs, true_labels)
+        if n_classes > 1:
+            expected_value = independent_metrics_dict[key](predicted_probs, true_labels.squeeze())
```

Comment:
I feel like a `squeeze()` shouldn't be necessary in this case. If it is, it suggests the shapes could be handled more cleanly upstream. I'd be in favour of making the shapes consistent everywhere, and special-casing on `n_classes` only where strictly necessary. This should make our lives easier by limiting code complexity and reducing chance of bugs. What do you think?

Reply:
This is also related to the other comment in `deepmil.py`.

Reply:
I agree, I have tried to limit the special-casing on `n_classes`